### DISCUSSION TITLE
Porting guide

### DISCUSSION THREAD

**Discussion by alejoseb** [berry-lang/berry#234] [Q&A]
Hi,
I tested Berry on Tasmota and it is awesome.  

I saw the example for STM32 with a makefile, but it is not working anymore. 

How I can port the current version?

Particularly on the documentation (https://github.com/berry-lang/berry/wiki/Appendix-B#2-porting)
Can anybody elaborate a bit more please?
What is the meaning of "Use map_build tool to generate constant object code and then compile"

I would like to integrate directly berry in a ST Cube IDE project with gcc compiler.

Thanks

**Comment by s-hadinger** [comment]
Hi, I suggest to not use `BE_USE_PRECOMPILED_OBJECT` when starting to port Berry, it makes things simpler.

The porting process is very straightforward, you need to adapt `be_port.c` (even here you can do the file part later), and everything else should just compile fine.

Did you try compiling the source?

**Reply by alejoseb** [reply]
Hi, thanks for your answer. 
Yes I did it, but I  used  BE_USE_PRECOMPILED_OBJECT  = 1 as default, otherwise code did not compile. 
Besides that, I modified be_port.c as you suggested and I am able to interact with the Berry's REPL using a serial console connected to my STM32 microcontroller.
However, I am not sure why it is not printing numbers, it only prints strings. When I try to print a number I get `ld`. I am not sure if I am porting something incorrectly.  You can see on the attached image that I am able to print the string variable `a`, but I cannot print the integer variable `b`.  The same code works as expected on Linux though.

![berry](https://user-images.githubusercontent.com/16860024/160715374-fd8a9092-cf17-4de6-b516-ca534c376601.png)
